plugins/package-managers/pub/src/main/kotlin/Lockfile.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.pub
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.yamlMapper
+
+internal fun parseLockfile(lockfile: File) = yamlMapper.readValue<Lockfile>(lockfile)
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+internal data class Lockfile(
+    val packages: Map<String, PackageInfo> = emptyMap()
+)
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+internal data class PackageInfo(
+    val dependency: String,
+    val description: Description,
+    val source: String? = null,
+    val version: String? = null
```

Comment:
Is the version of a package really optional in a lock file? I believe in the original code `pkgInfoFromLockfile["version"].textValueOrEmpty()` was used only because of generic map lookup.

Reply:
I've changed it for `version`, but left it alone for the other properties, because for the version this really seems
to be not nullable while for the others its IMO less clear and I do not want to do much functionality change in this transformational commit.

Reply:
I've reverted back to `nullable`, because according to `PubCacheReaderTest` the `version` is omit for local dependencies.